db/model/subject.js

```diff
@@ -344,8 +344,19 @@ module.exports = function subject(seq, dataTypes) {
 
         if (inst.getDataValue('isPublished')) {
           if (inst.previous('isPublished')) {
-            common.publishChange(inst, eventName.upd, changedKeys,
+            /*
+             * If tags were updated, send a "delete" event followed by an "add"
+             * event so that perspectives using subject tag filters will get
+             * the right realtime events. If subject tags were not updated,
+             * just send the usual "update" event.
+             */
+            if (inst.changed('tags')) {
+              common.publishChange(inst, eventName.del);
```

Comment:
Please add a TODO comment here describing the difficulties in writing unit tests for this right now.